react-demo-app/src/components/Video.tsx

```diff
@@ -0,0 +1,184 @@
+        }));
+
+        setupSessionListenersForVideo(session);
+      }
+    });
+  }, []);
+
+  function startVideoConf() {
+    sdk.startVideoConference(roomJid);
+  }
+
+  function startVideoMeeting() {
+    sdk.startVideoMeeting(roomJid);
+  }
+
+  function startScreenShare() {
+    if (!sessionState?.startScreenShare) {
+      return;
+    }
+    sessionState.startScreenShare();
+  }
+
+  async function startMedia() {
+    return await sdk.media.startMedia({video: true, audio: false});
+  }
+
+  function logRelevantSessionEvents(session: IExtendedMediaSession) {
+    const sessionEventsToLog: Array<keyof SessionEvents> = ['participantsUpdate', 'activeVideoParticipantsUpdate', 'speakersUpdate'];
+    sessionEventsToLog.forEach((eventName) => {
+      session.on(eventName, (e) => console.info(eventName, e));
+    });
+  }
+
+  function setupSessionListenersForVideo(session: VideoMediaSession) {
+
+    session.on('incomingMedia', () => {
+      setIsIncomingStreamActive(true);
+      if (vanityVideoRef.current && session?._outboundStream) {
+        vanityVideoRef.current.srcObject = session._outboundStream;
+      }
+    });
+
+    session.on('sessionState', () => {
+      dispatch(forceVideoConversationUpdate({ conversationId: session.conversationId }));
+    });
+
+    session.on('connectionState', () => {
+      dispatch(forceVideoConversationUpdate({ conversationId: session.conversationId }));
+    });
+
+    session.on('participantsUpdate', partsUpdate => {
+      const participant = partsUpdate.activeParticipants.find(part => part.userId === session.fromUserId);
+      const conversation = videoConversations.find(c => c.conversationId === partsUpdate.conversationId);
+      const participantStore = conversation?.participantsUpdate?.activeParticipants?.find(p => p.userId === session.fromUserId);
+      if (participant?.audioMuted !== participantStore?.audioMuted) {
+        dispatch(updateAudioLoading({loading: false, convId: partsUpdate.conversationId}));
+      }
+      if (participant?.videoMuted !== participantStore?.videoMuted) {
+        dispatch(updateVideoLoading({loading: false, convId: partsUpdate.conversationId}));
+      }
+
+      dispatch(addParticipantUpdateToVideoConversation(partsUpdate));
+      setIsIncomingStreamActive(partsUpdate.activeParticipants.length < 2 ? false :
+        !partsUpdate.activeParticipants.find(part => session.fromUserId !== part.userId)?.videoMuted);
+      setIsOutgoingStreamActive(
+        !partsUpdate.activeParticipants.find(part => session.fromUserId === part.userId)?.videoMuted);
+    });
+
+    session.on('terminated', reason => {
+      dispatch(removeVideoConversationFromActive({conversationId: session.conversationId, reason: reason}));
+      setIsIncomingStreamActive(false);
+      setIsOutgoingStreamActive(false);
+    });
+  }
+
+  return (
+    <>
+      <Card className="video-card">
+        <h2>Video</h2>
+        <div className="buttons-and-video-container">
```

Comment:
Can we extract this into a card and copy similar styling as the "Place a Call" card for softphone?